ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -423,19 +436,29 @@ func (r *RayJobReconciler) releaseComputeResources(ctx context.Context, rayJobIn
 		job := &batchv1.Job{}
 		if err := r.Client.Get(ctx, client.ObjectKey{Namespace: namespace, Name: jobName}, job); err != nil {
 			if errors.IsNotFound(err) {
+				isJobNotFound = true
 				r.Log.Info("The submitter Kubernetes Job has been already deleted", "RayJob", rayJobInstance.Name, "Kubernetes Job", job.Name)
-				return nil
 			} else {
 				r.Log.Error(err, "Failed to get Kubernetes Job")
-				return err
+				return false, err
+			}
+		} else {
+			if !job.DeletionTimestamp.IsZero() {
+				r.Log.Info("The Job deletion is ongoing.", "RayJob", rayJobInstance.Name, "Submitter K8s Job", job.Name)
+			} else {
+				if err := r.Client.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationBackground)); err != nil {
+					return false, err
+				}
+				r.Log.Info("The associated submitter Job is deleted", "RayJob", rayJobInstance.Name, "Submitter K8s Job", job.Name)
+				r.Recorder.Eventf(rayJobInstance, corev1.EventTypeNormal, "Deleted", "Deleted submitter K8s Job %s", job.Name)
 			}
-		}
-		if err := r.Client.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationBackground)); err != nil {
-			r.Log.Error(err, "Failed to delete the Kubernetes Job")
-			return err
 		}
 	}
-	return nil
+
+	isReleaseComplete := isClusterNotFound && isJobNotFound
```

Comment:
Without this PR, this function returns nil when the DeletionTimeStamp is not zero.